test/machineslib.py

```diff
@@ -408,8 +412,50 @@ def setUp(self):
         # HACK: older c-ws versions always log an assertion, fixed in PR cockpit#16765
         self.allow_journal_messages("json_object_get_string_member: assertion 'node != NULL' failed")
 
+    def downloadVmXml(self, vm):
+        m = self.machine
+
+        vms_xml = f"/var/lib/cockpittest/pkg_specific_artifacts/{vm}.xml"
+        m.execute(f"virsh dumpxml {vm} > {vms_xml}")
+        dest_file = f"{self.label()}-{m.label}-{vm}.xml"
+        dest = os.path.abspath(dest_file)
+        m.download(vms_xml, dest)
+        attach(dest, move=False)
+        print(f"Wrote {vm} XML to {dest_file}")
+
+    def downloadVmLog(self, vm):
+        m = self.machine
+
+        vms_log = f"/var/log/libvirt/qemu/{vm}.log"
+        # Log file may not exist, if VM was never started
+        if m.execute(f"!test -f {vms_log} || echo exists") == "exists":
```

Comment:
[This](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1158-20230731-093210-86f4fc47-fedora-38-pybridge/log.html#64) shows why:

   bash: line 1: !test: command not found

/me buys a space